krux/logging.py

```diff
@@ -121,6 +122,25 @@ def get_logger(name, **kwargs):
 
     All other keywords are passed verbatim to the setup() function.x
     """
-    setup(**kwargs)
-
-    return logging.getLogger(name)
+    if syslog_facility is None:
+        setup(**kwargs)
+        return logging.getLogger(name)
+    else:
+        assert syslog_facility in logging.handlers.SysLogHandler.facility_names, 'Invalid syslog facility %s' % syslog_facility
+        logger = logging.getLogger(name)
+        # On Linux, Python defaults to logging to localhost:514; on Ubuntu, rsyslog is not configured 
+        # to listen on the network. On other platforms (Darwin/OS X at least), Python by default sends
+        # to syslog vi a method by which syslog is listening.
+        if platform.system() == 'Linux':
+            handler = logging.handlers.SysLogHandler('/dev/log', facility=syslog_facility)
+        else:
+            handler = logging.handlers.SysLogHandler(facility=syslog_facility)
+        logger.addHandler(handler)
+        # set the level, if it was passed:
+        if 'level' in kwargs:
+            logger.setLevel(LEVELS[kwargs['level']])
+
+        # the default formatter munhges that tag for some reason
+        formatter = logging.Formatter('%(name)s: %(message)s')
+        handler.setFormatter(formatter)
+        return logger
```

Comment:
I would prefer we do syslog in *addition* to stdout/stderr, rather than *instead* of. We can discuss this further but the basic overview is that the stdlib is to represent the One True Way to do things, so it is not supposed to provide a bunch of options & choices. It is supposed to implement what we expect to see. If we add an option to log to syslog instead, we have to think about "is this app a syslog app or a stdout/stderr app". Instead we should just do both. Arguably, I guess we could switch all apps to syslog only, but I'd need to be convinced.